src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -388,40 +366,38 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
         }
 
         public async Task SwitchToAvailablePackagesAsync() {
-            await _coreShell.SwitchToMainThreadAsync();
-            if (_selectedTab == SelectedTab.AvailablePackages) {
-                return;
-            }
-
-            _selectedTab = SelectedTab.AvailablePackages;
-            IsLoading = !_availableLock.IsCompleted;
-
-            await EnsureAvailablePackagesLoadedAsync();
-            if (_selectedTab == SelectedTab.AvailablePackages) {
-                IsLoading = false;
-                ReplaceItems(_availablePackages);
+            if (await SetTabAsync(SelectedTab.AvailablePackages)) {
```

Comment:
>SetTabAsync [](start = 22, length = 11)

Please add `await _coreShell.SwitchToMainThreadAsync();` at the beginning of this method